popup/component/app.js

```diff
@@ -58,35 +65,65 @@ export default class App extends Component {
 
   render () {
     const { store } = this.props;
-    const { enabled, extractions } = this.state;
+    const { augmentationEnabled, extractions, status, url } = this.state;
 
     return (
       <div className={ styles.container }>
         <div className={ styles.header }>
           <h1 className={ styles.title }>Web3 Injection</h1>
-          <CheckBox
-            checked={ enabled }
-            className={ styles.check }
-            onChange={ this.handleChange }
-          />
         </div>
 
-        <Extractions
-          extractions={ extractions }
-          store={ store }
-        />
+        { this.renderExtractions(augmentationEnabled, extractions, store) }
+
+        { this.renderStatus(status, url) }
       </div>
     );
   }
 
-  @bind
-  handleChange (event) {
-    const { checked } = event.target;
+  renderExtractions (augmentationEnabled, extractions, store) {
+    if (!augmentationEnabled) {
+      return null;
+    }
 
-    Config.set({ enabled: checked })
-      .then(() => {
-        this.setState({ enabled: checked });
-      });
+    return (
+      <Extractions
+        extractions={ extractions }
+        store={ store }
+      />
+    );
+  }
+
+  renderStatus (status, url) {
+    const iconClassName = classnames({
+      [ styles.statusIcon ]: true,
+      [ styles.connected ]: status === 'connected',
+      [ styles.connecting ]: status === 'connecting',
+      [ styles.disconnected ]: status === 'disconnected'
+    });
+
+    let phrase;
+
+    switch (status) {
+      case 'connected':
+        phrase = 'Connected to';
+        break;
+
+      case 'connecting':
+        phrase = 'Connecting to';
+        break;
+
+      case 'disconnected':
+      default:
+        phrase = 'Not connected to';
+        break;
+    }
+
+    return (
+      <div className={ styles.status }>
+        <span className={ iconClassName } />
+        <span>{ phrase } { url }</span>
```

Comment:
Maybe it should also hint consequences when you are not connected to the node, like: "Web3 websites will not work.".